frontend/src/devicelist/SingleTableRow.jsx

```diff
@@ -0,0 +1,88 @@
+import React from "react";
+import {
+  Box,
+  Collapse,
+  IconButton,
+  TableCell,
+  TableRow,
+  Typography
+} from "@material-ui/core";
+
+import { ExpandLess, ExpandMore } from "@material-ui/icons/";
+
+import PropTypes from "prop-types";
+import ActionMenu from "./ActionMenu";
+import StatusIndicator from "../StatusIndicator";
+import ChannelDetailsTable from "./ChannelDetailsTable";
+
+export default class SingleTableRow extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false
+    };
+  }
+
+  render() {
+    const {
+      deviceDetails: { id, name, serialNumber, status, ip, channels }
+    } = this.props;
+    const { open } = this.state;
+    return (
+      <>
+        <TableRow key={id} className="singleDeviceRow">
+          <TableCell
+            className="tableCell dropdownButton"
+            style={{ width: 1, padding: 0, paddingLeft: 5 }}
+          >
+            <IconButton onClick={() => this.setState({ open: !open })}>
+              {open ? <ExpandLess /> : <ExpandMore />}
+            </IconButton>
+          </TableCell>
+          <TableCell className="tableCell">{name}</TableCell>
+          <TableCell className="tableCell numeric">{serialNumber}</TableCell>
+          <TableCell className="tableCell">
+            <StatusIndicator status={status} />
+          </TableCell>
+          <TableCell className="tableCell numeric">{ip}</TableCell>
+          <TableCell className="tableCell" align="center">
+            <ActionMenu />
+          </TableCell>
+        </TableRow>
+        <TableRow className="deviceDetails">
+          <TableCell className="chevronText lightestGrey" colSpan={7}>
+            <Collapse in={open} timeout="auto" unmountOnExit>
```

Comment:
Not quite sure what was changed here exactly. but from what I can see, not much. However, in terms of end result The CSS of the element seems to have changed a lot. with a lot of new properties that all seem to originate from actually using the mui table root class styling. This is more of a note for the cleanup PR but I tried

![image](https://user-images.githubusercontent.com/35614138/100392841-cdc2d680-3005-11eb-9330-014185ebc16d.png)
![image](https://user-images.githubusercontent.com/35614138/100393140-a91b2e80-3006-11eb-8701-1f69a6889f7b.png)

Reply:
I changed the invalid DOM property 'class' to 'className'

Reply:
not having the class at all removes all the styling, which isn't great It will require some investigation so, feel free to leave this for a later PR